backbone.js

```diff
@@ -14,24 +14,24 @@
 
   // Set up Backbone appropriately for the environment. Start with AMD.
   if (typeof define === 'function' && define.amd) {
-    define(['underscore', 'jquery', 'exports'], function(_, $, exports) {
+    define(['underscore', 'jquery', 'promise', 'exports'], function(_, $, Promise, exports) {
       // Export global even in AMD case in case this script is loaded with
       // others that may still expect a global Backbone.
-      root.Backbone = factory(root, exports, _, $);
+      root.Backbone = factory(root, exports, _, $, Promise);
     });
 
   // Next for Node.js or CommonJS. jQuery may not be needed as a module.
   } else if (typeof exports !== 'undefined') {
-    var _ = require('underscore'), $;
+    var _ = require('underscore'), $, Promise = require('promise');
     try { $ = require('jquery'); } catch(e) {}
-    factory(root, exports, _, $);
+    factory(root, exports, _, $, Promise);
 
   // Finally, as a browser global.
   } else {
-    root.Backbone = factory(root, {}, root._, (root.jQuery || root.Zepto || root.ender || root.$));
+    root.Backbone = factory(root, {}, root._, (root.jQuery || root.Zepto || root.ender || root.$), root.Promise);
   }
 
-}(function(root, Backbone, _, $) {
+}(function(root, Backbone, _, $, Promise) {
```

Comment:
Yes... if your project didn't include the `promise` library this bit of code would fail. It's much better to require the dev to provide a Promise polyfill or set one on `Backbone.Promise` directly instead.